### PR TITLE
Add Undo Feature for Edit and Clear

### PR DESCRIPTION
**Pull request by garylyp**
Allow Edit and Clear commands to be undone or redone. 
Issue #62

### CODE REVIEW

File: src/main/java/seedu/exercise/logic/commands/ClearCommand.java [hunk 1]
```diff
@@ -2,22 +2,31 @@
...
         requireNonNull(model);
+        exerciseBookPrevious = new ExerciseBook(model.getAllExerciseData()); // Create a deep copy
         model.setExerciseBook(new ExerciseBook());
+        EventHistory.getInstance().addCommandToUndoStack(this);
         return new CommandResult(MESSAGE_SUCCESS);
     }
+
+    public ReadOnlyExerciseBook getExerciseBookPrevious() {
+        return exerciseBookPrevious;
+    }
```
Comment (t-cheepeng) [inline src/main/java/seedu/exercise/logic/commands/ClearCommand.java]: Should the command itself keep the previous exercise book? Seems like it violates SRP since ClearCommand sole responsibility is to execute the clear command by the user.

Maybe move the `exerciseBookPrevious` into the `EventHistory` since it is used there

Comment (t-cheepeng) [reply src/main/java/seedu/exercise/logic/commands/ClearCommand.java]: Same for the remaining undoable commands

Comment (garylyp) [reply src/main/java/seedu/exercise/logic/commands/ClearCommand.java]: I have decided to stick to the current design because I am truly unable to find simpler solutions that would not introduce unnecessary and unjustifiable dependencies for EventHistory and EventFactory classes.

To justify the storage of the previous exercise book in ClearCommand, I am renaming the variable from ExerciseBookPrevious to ExerciseBookCleared to indicate that this ExerciseBookCleared is unique to the ClearCommand, like how the ExerciseDeleted is unique to the DeleteCommand. This does make a bit of sense in showing that this ClearCommand was particularly involved in clearing this version of the exercise book.

TLDR: Really need to store the older version of ExerciseBook in ClearCommand to access it in the EventFactory when producing the ClearEvent.
